src/hotspot/share/jfr/leakprofiler/checkpoint/rootResolver.cpp

```diff
@@ -249,7 +249,8 @@ bool ReferenceToThreadRootClosure::do_thread_stack_detailed(JavaThread* jt) {
 
   if (jt->has_last_Java_frame()) {
     // Traverse the monitor chunks
-    MonitorChunk* chunk = jt->monitor_chunks();
+    // Consider removing, chunk will always be null.
+    MonitorChunk* chunk = jt->monitor_chunks_safe();
```

Comment:
This seems like a good candidate for removal, to be replaced by an assert.  If we can walk the stack below, then either "jt" is the current thread, or we are at a safepoint.